### PR TITLE
[SPARK-49843][SQL][3.4] Fix change comment on char/varchar columns 

### PR DESCRIPTION
**Pull request by stefankandic**
### What changes were proposed in this pull request?

Fix the issue in `AlterTableChangeColumnCommand` where changing the comment of a char/varchar column also tries to change the column type to string. 

Backporting #48315 to 3.4

### Why are the changes needed?

Because the newColumn will always be a `StringType` even when the metadata says that it was originally char/varchar.

### Does this PR introduce _any_ user-facing change?

Yes, the query will no longer fail when using this code path.

### How was this patch tested?

New query in golden files.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

**Review by MaxGekk — APPROVED** [review]
Waiting for CI.

**Comment by stefankandic** [comment]
@MaxGekk These OOM failures don't really seem relate to my changes. Do you know anything about this?

**Comment by MaxGekk** [comment]
@stefankandic The commits in `branch-3.4` are all green: https://github.com/apache/spark/commits/branch-3.4/ . Let's re-rung failed GAs.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM (Pending CIs.)

Could you re-trigger the CIs, @stefankandic ?

**Comment by dongjoon-hyun** [comment]
Merged to branch-3.4. Thank you, @stefankandic and @MaxGekk .

**Pull request closed without merging**
